x-pack/lib/filters/geoip/database_manager.rb

```diff
@@ -187,6 +225,39 @@ def trigger_download
     end
   end
 
+  def initialize_metrics
+    metadatas = @metadata.get_all
+    metadatas.each do |row|
+      type = row[DatabaseMetadata::Column::DATABASE_TYPE]
+      @metric.namespace([:database, type.to_sym]).tap do |n|
+        n.gauge(:status, @states[type].is_eula ? :healthy : :init)
+        if @states[type].is_eula
+          n.gauge(:download_at, unix_time_to_iso8601(row[DatabaseMetadata::Column::DIRNAME]))
+          n.gauge(:fail_check_in_days, time_diff_in_days(row[DatabaseMetadata::Column::CHECK_AT]))
+        end
+      end
+    end
+
+    @metric.namespace([:download]).tap do |n|
+      check_at = metadatas.map { |row| row[DatabaseMetadata::Column::CHECK_AT].to_i }.max
+      n.gauge(:last_check_at, unix_time_to_iso8601(check_at))
+    end
+  end
+
+  def set_download_metric(success_cnt)
+    @metric.namespace([:download]).tap do |n|
+      n.gauge(:last_check_at, Time.now.iso8601)
+
+      if success_cnt == DB_TYPES.size
+        n.increment(:successes, 1)
+        n.gauge(:status, :succeeded)
+      else
+        n.increment(:failures, 1)
+        n.gauge(:status, :failed)
+      end
+    end
```

Comment:
Maybe this could be split in two methods, one for the counter and the other for the status, so that the status update points becomes evident in the download flow